internal/events/stream/event_test.go

```diff
@@ -0,0 +1,60 @@
+package stream
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	eventstream "github.com/openshift/assisted-service/pkg/stream"
+)
+
+type TestStruct struct {
+	Foo string `json:"foo"`
+	Bar string `json:"bar"`
+}
+
+var _ = Describe("Stream event", func() {
+	var (
+		stream *eventstream.MockEventStreamWriter
+		ctrl   *gomock.Controller
+	)
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		stream = eventstream.NewMockEventStreamWriter(ctrl)
+	})
+
+	It("return error if stream is not configured", func() {
+
+		res := StreamEvent(nil, "TestEvent", "mykey", nil)
+		Expect(res).ShouldNot(BeNil())
+	})
+
+	It("sends event with regular payload", func() {
+
+		payload := &TestStruct{
+			Foo: "bar",
+			Bar: "foo",
+		}
+		expectedPayload := []byte(`{"name":"TestEvent","payload":{"foo":"bar","bar":"foo"},"metadata":{"versions":{"assisted-installer":"quay.io/edge-infrastructure/assisted-installer:latest","assisted-installer-controller":"quay.io/edge-infrastructure/assisted-installer-controller:latest","assisted-installer-service":"Unknown","discovery-agent":"quay.io/edge-infrastructure/assisted-installer-agent:latest"}}}`)
+		stream.EXPECT().Write([]byte("mykey"), expectedPayload).Times(1).Return(nil)
+		res := StreamEvent(stream, "TestEvent", "mykey", payload)
+		Expect(res).To(BeNil())
+	})
+
+	It("sends event with nil payload", func() {
+
+		var payload map[string]string
+
+		expectedPayload := []byte(`{"name":"TestEvent","payload":null,"metadata":{"versions":{"assisted-installer":"quay.io/edge-infrastructure/assisted-installer:latest","assisted-installer-controller":"quay.io/edge-infrastructure/assisted-installer-controller:latest","assisted-installer-service":"Unknown","discovery-agent":"quay.io/edge-infrastructure/assisted-installer-agent:latest"}}}`)
+		stream.EXPECT().Write([]byte("mykey"), expectedPayload).Times(1).Return(nil)
+		res := StreamEvent(stream, "TestEvent", "mykey", payload)
+		Expect(res).To(BeNil())
+	})
+
+})
+
+func TestStream(t *testing.T) {
```

Comment:
don't we miss here validating the metadata for versions-info?

Reply:
We are validating it right? Are you suggesting to change the version and validate that it actually changes in the metadata? Wouldn't this better be tested inside the versions package instead?

Reply:
no you're right, it's already validated 
I missed it because ``expectedPayload`` is such a blob...